lib/msal-node/src/network/HttpClient.ts

```diff
@@ -4,424 +4,200 @@
-                    parsedHeaders,
-                    parseBody(
-                        httpStatusCode,
-                        statusMessage,
-                        parsedHeaders,
-                        body
-                    ) as T,
-                    httpStatusCode
-                );
-
-                if (
-                    (httpStatusCode <
-                        CommonConstants.HTTP_SUCCESS_RANGE_START ||
-                        httpStatusCode >
-                            CommonConstants.HTTP_SUCCESS_RANGE_END) &&
-                    // do not destroy the request for the device code flow
-                    networkResponse.body["error"] !==
-                        Constants.AUTHORIZATION_PENDING
-                ) {
-                    request.destroy();
-                }
-                resolve(networkResponse);
-            });
-
-            socket.on("error", (chunk) => {
-                request.destroy();
-                socket.destroy();
-                reject(new Error(chunk.toString()));
-            });
-        });
-
-        request.on("error", (chunk) => {
-            request.destroy();
-            reject(new Error(chunk.toString()));
-        });
-    });
-};
-
-const networkRequestViaHttps = <T>(
-    urlString: string,
-    httpMethod: string,
-    options?: NetworkRequestOptions,
-    agentOptions?: https.AgentOptions,
-    timeout?: number
-): Promise<NetworkResponse<T>> => {
-    const isPostRequest = httpMethod === HttpMethod.POST;
-    const body: string = options?.body || "";
-
-    const url = new URL(urlString);
-    const headers = options?.headers || ({} as Record<string, string>);
-    const customOptions: https.RequestOptions = {
-        method: httpMethod,
-        headers: headers,
-        ...NetworkUtils.urlToHttpOptions(url),
-    };
-
-    if (agentOptions && Object.keys(agentOptions).length) {
-        customOptions.agent = new https.Agent(agentOptions);
-    }
-
-    if (isPostRequest) {
-        // needed for post request to work
-        customOptions.headers = {
-            ...customOptions.headers,
-            "Content-Length": body.length,
-        };
-    } else {
-        // optional timeout is only for get requests (regionDiscovery, for example)
-        if (timeout) {
-            customOptions.timeout = timeout;
-        }
-    }
+            if (error instanceof Error && error.name === "AbortError") {
+                throw new Error("Request timeout");
+            }
 
-    return new Promise<NetworkResponse<T>>((resolve, reject) => {
-        let request: http.ClientRequest;
-        // managed identity sources use http instead of https
-        if (customOptions.protocol === "http:") {
-            request = http.request(customOptions);
-        } else {
-            request = https.request(customOptions);
+            throw new Error(
+                `Network request failed: ${
+                    error instanceof Error ? error.message : String(error)
```

Comment:
Parsing an object of unknown shape is not safe, we should put a placeholder in here instead
```suggestion
                    error instanceof Error ? error.message : "unknown"
```

Reply:
done